### PR TITLE
feat: generic module deployment roles

### PR DESCRIPTION
**Pull request by kukushking**
*Issue #, if available:*
- https://github.com/awslabs/seed-farmer/issues/667

*Description of changes:*
- isolate module deployment role logic
- create generic module deployment roles during priming accounts & destroy at teardown
- update apply/destroy to use generic module deployment roles when no modulestack is present

NOTE: current PR does not include a cleanup of dangling module deployment roles from previous Seed-Farmer versions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: seedfarmer/commands/_deployment_commands.py [hunk 1]
```diff
@@ -80,6 +81,83 @@ def _process_data_files(data_files: List[DataFile], module_name: str, group_name
         raise seedfarmer.errors.InvalidPathError("Missing DataFiles - cannot process")
 
 
+def _get_generic_module_deployment_role_name(
+    project_name: str, deployment_name: str, account_id: str, region: str, qualifier: Optional[str] = None
+) -> str:
+    name = f"{project_name}-{deployment_name}-{account_id}-{region}-deployment-role"
```
Comment (dgraeber) [inline seedfarmer/commands/_deployment_commands.py:87]: shouldn't there be a check for length?

Comment (dgraeber) [reply seedfarmer/commands/_deployment_commands.py:87]: Why account id?  Aren't the roles segemented to an account?

File: seedfarmer/commands/_deployment_commands.py [hunk 2]
```diff
@@ -94,22 +172,31 @@ def _execute_deploy(
...
+            group_name=mdo.group_name,
+            module_name=mdo.module_name,
+            account_id=account_id,
+            region=region,
+            parameters=mdo.parameters,
+            docker_credentials_secret=mdo.docker_credentials_secret,
+            permissions_boundary_arn=mdo.permissions_boundary_arn,
+        )
+    else:
+        module_role_name = _get_generic_module_deployment_role_name(
```
Comment (dgraeber) [inline seedfarmer/commands/_deployment_commands.py:190]: should there be checks to make sure the role exists?

File: seedfarmer/commands/_deployment_commands.py [hunk 3]
```diff
@@ -273,15 +373,27 @@ def prime_target_accounts(
...
-            return [args["account_id"], args["region"], seedkit_stack_outputs]
+
+            deployment_role_name = create_generic_module_deployment_role(
+                account_id=target_account_id,
+                region=target_region,
+                deployment_manifest=deployment_manifest,
+                qualifier=qualifier,
+            )
+
+            seedkit_stack_outputs["DeploymentRoleName"] = deployment_role_name
```
Comment (dgraeber) [inline seedfarmer/commands/_deployment_commands.py:395]: ummm, why add to the seedkit outputs?  Should this be hung on the targetAccount Mapping object?

Comment (kukushking) [reply seedfarmer/commands/_deployment_commands.py]: good call, on it

**Review by dgraeber — CHANGES_REQUESTED** [review]

**Review by malachi-constant — APPROVED** [review]

**Review by dgraeber — APPROVED** [review]

**Pull request merged**